scripts/keras_classification.py

```diff
@@ -0,0 +1,738 @@
+        args.dataset,
+        root=args.dataset_path,
+        train=train,
+        image_size=image_size,
+        **kwargs,
+    )
+    LOGGER.info(
+        "created {} dataset: {}, images to resize to {}".format(
+            "train" if train else "val", dataset, image_size
+        )
+    )
+
+    # get num_classes
+    if args.dataset == "imagefolder":
+        num_classes = dataset.num_classes
+    else:
+        dataset_attributes = DatasetRegistry.attributes(args.dataset)
+        num_classes = dataset_attributes["num_classes"]
+
+    return dataset, num_classes
+
+
+def build_dataset(args, dataset: Dataset, train: bool = True) -> tf.data.Dataset:
+    test_batch_size = (
+        args.test_batch_size if args.test_batch_size else args.train_batch_size
+    )
+    batch_size = args.train_batch_size if train else test_batch_size
+    if train:
+        shuffle_buffer_size = (
+            args.train_shuffle_buffer_size
+            if args.train_shuffle_buffer_size
+            else dataset.num_images
+        )
+    else:
+        shuffle_buffer_size = None
+
+    if train:
+        prefetch_buffer_size = (
+            args.train_prefetch_buffer_size
+            if args.train_prefetch_buffer_size
+            else batch_size * 8
+        )
+    else:
+        prefetch_buffer_size = (
+            args.test_prefetch_buffer_size
+            if args.test_prefetch_buffer_size
+            else batch_size * 8
+        )
+    built_dataset = dataset.build(
+        batch_size,
+        repeat_count=1,
+        shuffle_buffer_size=shuffle_buffer_size,
+        prefetch_buffer_size=prefetch_buffer_size,
+        num_parallel_calls=args.dataset_parallel_calls,
+    )
+    return built_dataset
+
+
+def create_model(args, input_shape, num_classes):
+    kwargs = args.model_kwargs
+
+    if args.checkpoint_path == "zoo":
+        zoo_model = ModelRegistry.create_zoo_model(
+            args.arch_key, args.pretrained, args.pretrained_dataset
+        )
+        model_file_paths = zoo_model.download_framework_files(extensions=[".h5"])
+        if not model_file_paths:
+            model_file_paths = zoo_model.download_framework_files(extensions=[".tf"])
+        if not model_file_paths:
+            raise RuntimeError("Error downloading model from SparseZoo")
+        model_file_path = model_file_paths[0]
+        model = keras.models.load_model(model_file_path)
+    else:
+        model = ModelRegistry.create(
+            args.arch_key,
+            args.pretrained,
+            args.checkpoint_path,
+            args.pretrained_dataset,
+            input_shape=input_shape,
+            **kwargs,
+        )
+
+    if (
+        args.pretrained_dataset is not None
+        and args.dataset is not None
+        and args.pretrained_dataset != args.dataset
+    ):
+        # Set up the model for fine-tuning
+        model.trainable = False
+        new_output = keras.layers.Dense(num_classes, activation="softmax")(
```

Comment:
We can't force use of softmax here, need to figure out a more generic way to allow different activations such as sigmoid for multi class

Reply:
Add an argument for the transfer class_type, and optionally the name of the layer whose outputs to be connected to the new layer for transfer learning (default None for model.layers[-2])